Uploading initial idl files to set the framework for allowing Local network access

igower opened the pull request:
#### 89cda16d1500368ae359cf95b7739d89ec77e355
<pre>
Uploading initial idl files to set the framework for allowing Local network access
<a href="https://bugs.webkit.org/show_bug.cgi?id=295935">https://bugs.webkit.org/show_bug.cgi?id=295935</a>
<a href="https://rdar.apple.com/155825558">rdar://155825558</a>

Reviewed by Anne van Kesteren.

I uploaded the IDL files as specified int he draft spec and included an IPAddressSpace.h file to support the framework.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchRequest.h:
* Source/WebCore/Modules/fetch/FetchRequest.idl:
* Source/WebCore/Modules/fetch/FetchRequestInit.h:
(WebCore::FetchRequestInit::hasMembers const):
* Source/WebCore/Modules/fetch/FetchRequestInit.idl:
* Source/WebCore/Modules/fetch/IPAddressSpace.h: Copied from Source/WebCore/Modules/fetch/FetchRequestInit.idl.
* Source/WebCore/Modules/fetch/IPAddressSpace.idl: Copied from Source/WebCore/Modules/fetch/FetchRequestInit.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/297496@main">https://commits.webkit.org/297496@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e314a4e1214217ae96c65f89acbe8bc022dab11

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111956 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31626 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22112 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117978 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62190 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/113918 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32303 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40204 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85062 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35729 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114903 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25798 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100746 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65494 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25121 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18880 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61827 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95190 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18956 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121288 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38989 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29007 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93911 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39370 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96998 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93728 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38924 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16715 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35011 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18050 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38883 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44395 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38520 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41848 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40236 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/Modules/fetch/FetchRequest.idl:
```diff
@@ -55,6 +55,7 @@ typedef (FetchRequest or USVString) RequestInfo;
     // FIXME: Implement 'isHistoryNavigation'.
     // readonly attribute boolean isHistoryNavigation;
     readonly attribute AbortSignal signal;
+    readonly attribute IPAddressSpace? targetAddressSpace;
```
```suggestion
    [EnabledBySetting=LocalNetworkAccessEnabled] readonly attribute IPAddressSpace? targetAddressSpace;
```

annevk commented on Source/WebCore/Modules/fetch/FetchRequestInit.idl:
```diff
@@ -43,4 +43,5 @@ dictionary FetchRequestInit {
     any signal;
     RequestPriority priority;
     any window; // can only be set to null
+    IPAddressSpace targetAddressSpace;
```
```suggestion
    [EnabledBySetting=LocalNetworkAccessEnabled] IPAddressSpace targetAddressSpace;
```

annevk commented on Source/WebCore/Modules/fetch/IPAddressSpace.h:
```diff
@@ -0,0 +1,38 @@
+
+namespace WTF {
+class URL;
+}
```
I don't understand why this is needed here.

annevk commented on Source/WebCore/Modules/fetch/IPAddressSpace.h:
```diff
@@ -0,0 +1,38 @@
+
+namespace WebCore {
+
+enum class IPAddressSpace : uint8_t {
```
```suggestion
enum class IPAddressSpace : bool {
```

annevk commented on Source/WebCore/Modules/fetch/IPAddressSpace.h:
```diff
@@ -0,0 +1,34 @@
+    Public,
+    Local
+};
+} // namespace WebCore
```
Nit: newline between the braces.

annevk reviewed (approved):
Looks good modulo nit.

igower committed:
Uploading initial idl files to set the framework for allowing Local network access
https://bugs.webkit.org/show_bug.cgi?id=295935
rdar://155825558

Reviewed by Anne van Kesteren.

I uploaded the IDL files as specified int he draft spec and included an IPAddressSpace.h file to support the framework.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchRequest.h:
* Source/WebCore/Modules/fetch/FetchRequest.idl:
* Source/WebCore/Modules/fetch/FetchRequestInit.h:
(WebCore::FetchRequestInit::hasMembers const):
* Source/WebCore/Modules/fetch/FetchRequestInit.idl:
* Source/WebCore/Modules/fetch/IPAddressSpace.h: Copied from Source/WebCore/Modules/fetch/FetchRequestInit.idl.
* Source/WebCore/Modules/fetch/IPAddressSpace.idl: Copied from Source/WebCore/Modules/fetch/FetchRequestInit.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/297496@main
